core/liquivision.c

```diff
@@ -278,6 +278,15 @@ static void parse_dives (int log_version, const unsigned char *buf, unsigned int
 			algorithm = *(buf + ptr++);	// 0=ZH-L16C+GF
 			sample_count = array_uint32_le(buf + ptr);
 		}
+
+		if (sample_count == 0) {
+			fprintf(stderr, "DEBUG: sample count 0 - terminating parser\n");
+			break;
+		}
+		if (ptr + sample_count * 4 + 4 > buf_size) {
+			fprintf(stderr, "DEBUG: BOF - terminating parser\n");
+			break;
+		}
```

Comment:
so assuming these two are for the same problem... sample_count is 0 and ptr + 4 > buf_size
but why do you have sample_count in there again? I guess I understand even less about that file format :-)

Reply:
Buffer overflow occurs when sample_count is insanely big.

I could have both cases in single if statement

(sample_count == 0 || ptr + sample_count * 4 + 4 > buf_size)

but more verbose code might help in understanding the file format (or at least that is why I wrote it that way).

Reply:
thanks for the explanation. Seems safe enough to merge.